runelite-client/src/main/java/net/runelite/client/plugins/thieving/ChestRespawn.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019, whs
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.thieving;
+
+import java.time.Instant;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import net.runelite.api.coords.WorldPoint;
+
+@AllArgsConstructor
+@Getter(AccessLevel.PACKAGE)
+class ChestRespawn
+{
+	private final Chest chest;
+	private final WorldPoint worldPoint;
+	private final Instant startTime;
```

Comment:
Rather than saving start time, wouldn't saving end time be better? then you can get the duration until that time comes, rather than getting start, adding the duration, and checking that against now

Reply:
For mining, they have this code that customize the respawn time for special rocks (eg. mining guild rocks) so I believe the choice to use startTime is better because the creator of this object doesn't need to handle those logic. Seeing that this file is based on that I think leaving it in allow for future changes if needed.

Reply:
Again, the class is inspired by that, but are there any chests that do change dynamically? If not, it doesn't apply to chests and should therefore be implemented in the way that's better if durations are static